### ISSUE TITLE
"minishift ssh" doesn't behave like "docker-machine ssh" as mentioned in doc

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#339]
As per `minishift ssh` help, it should behave like [`docker-machine ssh`](https://docs.docker.com/machine/reference/ssh/) where one can run command on VM.

```
$ minishift ssh df -h
Log in to or run a command on a Minishift VM with SSH. This command is similar to 'docker-machine ssh'.

Usage:
  minishift ssh [flags]

Global Flags:
      --alsologtostderr value          log to standard error as well as files
      --log-flush-frequency duration   Maximum number of seconds between log flushes (default 5s)
      --log_backtrace_at value         when logging hits line file:N, emit a stack trace (default :0)
      --log_dir value                  If non-empty, write log files in this directory
.....
```

The help usage mention that one can use command on a Minishift VM which seems not working at the moment.

Looking at the ["minishift" code for `ssh` command](https://github.com/minishift/minishift/blob/master/cmd/minikube/cmd/ssh.go#L35-L41), it looks like it doesn't provide any mechanism to support it.

@jimmidyson Please confirm.

**Comment by budhrg** [comment]
Found that if we wrap the command to be run inside VM in double quotes then it works.

```
$ minishift ssh "df -h"
```

Need a documentation fix here.

**Comment by hferentschik** [comment]
You need to consider bash/shell type interpolation and command line parsing here. You need to pass the command as whole and hence you need the quotes. Wether you need single or double quote will depend on whether you want variable interpolation or not. Not quite sure how and if we need to address this.

**Comment by budhrg** [comment]
@hferentschik 

> Not quite sure how and if we need to address this.

IMO, it's will act as a handy tool to inspect VM quickly. Its similar to `vagrant ssh -c command`.

WDYT @praveenkumar @LalatenduMohanty @gbraad ?

**Comment by praveenkumar** [comment]
> It's will act as a handy tool to inspect VM quickly. Its similar to `vagrant ssh -c command`

Does we had this features prior to we shift to `oc cluster up` way? As long as we handle bash/shell type interpolation and command line parsing as @hferentschik told then we can have it features.

**Comment by budhrg** [comment]
> Does we had this features prior to we shift to oc cluster up way? As long as we handle bash/shell type interpolation and command line parsing as @hferentschik told then we can have it features.

@jimmidyson Could you please confirm?

**Comment by jimmidyson** [comment]
Yes `minishift ssh` always worked fine before.

I think the problem here is the `-h` in the original command that cobra is parsing as a flag to minishift.

Here's output from 0.9.0:

```bash
$ minishift ssh df    
Filesystem           1K-blocks      Used Available Use% Mounted on
tmpfs                  1845184    378240   1466944  20% /
tmpfs                  1025100         0   1025100   0% /dev/shm
/dev/sda1             18745336    171324  17583288   1% /mnt/sda1
cgroup                 1025100         0   1025100   0% /sys/fs/cgroup
/dev/sda1             18745336    171324  17583288   1% /mnt/sda1/var/lib/docker/aufs

$ minishift ssh -h
Log into or run a command on a machine with SSH; similar to 'docker-machine ssh'

Usage:
  minishift ssh [flags]

Global Flags:
      --alsologtostderr                  log to standard error as well as files
      --log-flush-frequency duration     Maximum number of seconds between log flushes (default 5s)
      --log_backtrace_at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log_dir string                   If non-empty, write log files in this directory (default "")
      --logtostderr                      log to standard error instead of files
      --show-libmachine-logs             Whether or not to show logs from libmachine.
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          log level for V logs
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging

$ minishift-linux-amd64 ssh -- df -h
Filesystem                Size      Used Available Use% Mounted on
tmpfs                     1.8G    369.4M      1.4G  20% /
tmpfs                  1001.1M         0   1001.1M   0% /dev/shm
/dev/sda1                17.9G    167.3M     16.8G   1% /mnt/sda1
cgroup                 1001.1M         0   1001.1M   0% /sys/fs/cgroup
/dev/sda1                17.9G    167.3M     16.8G   1% /mnt/sda1/var/lib/docker/aufs
```

Notice the shell standard `--` to separate flags from arguments.

**Comment by budhrg** [comment]
@jimmidyson 

> Notice the shell standard -- to separate flags from arguments.

Agree. And hence, added a documentation fix.

**Comment by hferentschik** [comment]
Merged via pull request #341
